cachi2/core/package_managers/pip.py

```diff
@@ -1619,52 +1651,148 @@ def _validate_provided_hashes(requirements: list[PipRequirement], require_hashes
                 raise PackageRejected(msg, solution=None)
 
 
-def _download_pypi_package(
-    requirement: PipRequirement,
-    pip_deps_dir: RootedPath,
-    pypi_url: str,
-    pypi_auth: Optional[AuthBase] = None,
-) -> dict[str, Any]:
+@dataclass
+class DistributionPackageInfo:
+    """
+    A class representing relevant information about a distribution package.
+
+    Attributes:
+        name (str): The name of the package.
+        version (str): The version of the package.
+        package_type (Literal): sdist or wheel.
+        path (Path): The path to the package file.
+        url (str): The URL of the package.
+        is_yanked (bool): Indicates whether the package is yanked.
+        pypi_checksums (set[ChecksumInfo]): Set of checksums reported by PyPI.
+        user_checksums (set[ChecksumInfo]): Set of checksums specified by the user.
+        checksums_to_verify (set[ChecksumInfo]): Set of checksums to verify.
```

Comment:
Nitpick: I don't think this is bringing any helpful info besides what already could be inferred by the attributes in the class (although this makes the attributes visible in the IDE popup if you hover the class). 

I'd probably just remove it anyways, but I'm fine if you think keeping it is worth it.